### ISSUE TITLE
Conversion behavior on return types

### ISSUE THREAD

**Issue by ro99** [h33p/cglue#2]
Hi, this is a really cool library, nice work!! 

I have one question regarding the behavior of cglue on the return types.

The library will make the necessary conversions on function signatures that have, say, a &str as a parameter, but does not on the return type. 

For instance, if I have a function like:

```
#[cglue_trait]
pub trait Something {
    fn foo<'a>(&'a self, name: &str) -> &'a str;
}
```
Here the system will complain, warning that _`extern` fn uses type `str`, which is not FFI-safe_.

But if I use CSliceRef<'a, u8> directly on the return, it will work. We have this same behavior with slices (e.g. &'a[u8]), where we can use it as input parameter, but not on the return. Is this expected behavior? 

I am having a hard time trying to return a vector of strings 😅 . I know that I can use something like `*const *const u8`, but I am not sure if this is possible using CGlue only?

Thank you!

**Comment by ro99** [comment]
For now I am using the following workaround:

https://gist.github.com/iskakaushik/1c5b8aa75c77479c33c4320913eebef6

**Comment by h33p** [comment]
Hey there, thanks for bringing this up! It is an oversight on my part, and I will see what I can do about that.

**Comment by h33p** [comment]
As for vector of strings, things become a bit tricky, since `Vec` by itself is not very safe to share around the FFI boundary. Specifically, given a (rather theoretical) scenario where multiple libraries were to use different global allocators. This could be solved by creating a CVec type that has the correct drop method attached, but there is no real safe way to return a Vec without expensive re-allocations.

**Comment by h33p** [comment]
I have solved the first issue of strs and slices in return types in https://github.com/h33p/cglue/commit/c8a607e68a851321a4bc288491e879e34d541bd2. This will soon be available in 0.2, and I will later backport it to 0.1.
